components/core/src/clp_s/search/StringLiteral.hpp

```diff
@@ -68,7 +68,7 @@ class StringLiteral : public Literal {
             m_string_type = LiteralType::VarStringT;
         }
 
-        if (m_v.find('*') != std::string::npos) {
+        if (m_v.find('*') != std::string::npos || m_v.find('?') != std::string::npos) {
```

Comment:
I know it's wrong in the original code as well, but could we fix this to handle escaped `*` and `?` properly as well? This should really be checking just for unescaped `*` or `?`, and ignoring them when they're escaped. E.g. `var\?` is just a variable string.

Reply:
Right